### ISSUE TITLE
Install failed

### ISSUE THREAD

**Issue by sdondley** [pplu/aws-sdk-perl#387]
running perl 5 v 22. Install fails on an Ubuntu machine. Test summary report:

```
Test Summary Report
-------------------
t/01_load.t                           (Wstat: 9 Tests: 65 Failed: 0)
  Non-zero wait status: 9
  Parse errors: No plan found in TAP output
t/10_responses.t                      (Wstat: 0 Tests: 9900 Failed: 0)
  TODO passed:   6364, 6370, 7896, 9579, 9586, 9671
t/18_mocked.t                         (Wstat: 0 Tests: 4007 Failed: 0)
  TODO passed:   91-143, 145-154, 156-173, 175-222, 224-262
                264-322, 324-330
Files=26, Tests=14789, 476 wallclock secs ( 1.28 usr  0.18 sys + 81.69 cusr  3.81 csys = 86.96 CPU)
Result: FAIL
```

**Comment by castaway** [comment]
Hi, have you got a fuller report? ```prove -vl t``` would be helpful. My first guess would else be "missing dependencies"..

**Comment by sdondley** [comment]
Here is the entire build log.

[build.log](https://github.com/pplu/aws-sdk-perl/files/4455990/build.log)

**Comment by castaway** [comment]
That looks like you're missing https://metacpan.org/pod/Future, which isnt a direct dependency, but is a dep of https://metacpan.org/pod/Future::Mojo which we do use.. Redo dependency installing?

**Comment by sdondley** [comment]
I used cpanm to do the install. Tried it twice with cpanm, actually.

I've already forced installed Paws. I'll try on another machine when I get a chance.

**Comment by sdondley** [comment]
FYI, I just did `cpanm Future` on the machine and it installed without issue.

**Comment by castaway** [comment]
oh! 0.41 (I looked in 0.42) .. tried the newer one? 

hmm, Future::Mojo is only "suggest"ed, so its not that (despite erroring).. I assume its the fact that 01_load.t doesn't have an "ok" at the end, despite all subtests passing

Really would be useful if you could try either 0.42, or prove -vl t/01_load.t , when you get a moment, please

**Comment by sdondley** [comment]
Sorry, I'm not following. cpanm should install version 0.42 for me, shouldn't it? I tried installing Paws just a couple of days ago and last release was back on March 2nd.

**Comment by sdondley** [comment]
When I run `cpanm Paws` it reports version 0.41. Very weird. Maybe because I'm running an older version of Perl v 5.22 sub 1?

**Comment by castaway** [comment]
yes cpanm should find 0.42 as its definitely there! The version of perl required is still set to 5.10 or so. Is it using a local mirror?

**Comment by sdondley** [comment]
No idea. This is an amazon lightsail instance. I'm not very familiar with how they set their servers up. How do I check for a local mirror?

**Comment by sdondley** [comment]
See https://perlmonks.org/?node_id=324763

**Comment by Grinnz** [comment]
The currently indexed version of Paws is 0.41: https://cpanmeta.grinnz.com/packages?module=Paws&match_mode=exact

The uploader should have received a message on whether indexing succeeded or failed on the 0.42 upload, and can request a reindex on PAUSE here: https://pause.perl.org/pause/authenquery?ACTION=reindex

**Comment by Grinnz** [comment]
As for installing, you can work around this by installing with `cpanm Paws@0.42`

**Comment by castaway** [comment]
Im seeing 0.42 here: https://metacpan.org/release/Paws

**Comment by Grinnz** [comment]
Yes it got indexed as latest on metacpan, which tells me it was probably indexed in PAUSE at some point, but it no longer is (rollback? bug?). The PAUSE index is what CPAN installers use.

**Comment by sdondley** [comment]
Intalling with `cpanm Paws@0.42` failed as well: 
[build.log](https://github.com/pplu/aws-sdk-perl/files/4457307/build.log)

**Comment by castaway** [comment]
Have nudged @shadowcat-mst for ideas on why the build is failing, as he's more of an expert on the toolchain. Hopefully this'll help both with your install, and the missing 0.42 on cpan..

**Comment by sdondley** [comment]
OK, thanks. No rush, though. I force installed and it's working perfectly for my purposes so far.

**Comment by realflash** [comment]
I've just tried to install 0.43 from CPAN and it also fails because Future is missing. I also get

```
Skipping Paws::Net::FurlCaller due to probem loading: Can't locate Furl.pm
```

so perhaps someone could look at that in the process of fixing this.

[Build log](https://gist.github.com/realflash/19524697eba53d0b8625beda0561242b)

**Comment by realflash** [comment]
After I manually installed Future, it then failed due to the absence of Future:Mojo

[Seoncd build.log](https://gist.github.com/realflash/7abcad2ec482d650afd75f33607a443a)

**Comment by realflash** [comment]
I suspect there is more going on here than the missing pre-reqs. After manually installing Furl and Future and Future::Mojo, the install still fails, now with no dependencies missing. Additionally, there are no failed tests. Could it be to do with the test parse errors?

[Still broken after missing depndencies installed build.log](https://gist.github.com/realflash/9b1d1d966d44eb1c6587b76f37de89fa)

**Comment by pplu** [comment]
@realflash : Looks like the problem is:
```
t/10_responses.t                      (Wstat: 9 Tests: 8013 Failed: 0)
  TODO passed:   6364, 6370, 7896
  Non-zero wait status: 9
  Parse errors: No plan found in TAP output
```
It is exiting non-zero. Would you be able to send the last lines of the output of t/10_responses.t, to see why it is exiting non-zero?

**Comment by realflash** [comment]
```
ok 9899 - Got Summaries.1.HasFlaggedResources eq 0 from result
ok 9900 - Got Summaries.1.Timestamp eq 2015-10-14T10:40:20Z from result
1..9900
ok
All tests successful.

Test Summary Report
-------------------
t/10_responses.t (Wstat: 0 Tests: 9900 Failed: 0)
  TODO passed:   6364, 6370, 7896, 9579, 9586, 9671, 9736
Files=1, Tests=9900, 13 wallclock secs ( 0.76 usr  0.26 sys + 11.59 cusr  0.33 csys = 12.94 CPU)
Result: PASS
```

I then did a `prove -vl` and all 15013 tests passed. This was running inside the temp build folder mentioned in the third log. 

Then I did cpanm --test-only and it passed (although that's after force installing the package). Might need to reproduce it on a clean box (Ubuntu 20.04.2 server aarch64).
